core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/codeGen/CodeGeneratorImpl.kt

```diff
@@ -62,70 +64,82 @@ internal interface TypeRenderingStrategy {
+            is FieldType.ValueFieldType ->
+                fieldType.typeFqName
+
+            is FieldType.GroupFieldType ->
+                if (asAccessor) "$DataRow<${fieldType.markerName}>"
+                else fieldType.markerName
+
             is FieldType.FrameFieldType ->
-                "${DataFrame::class.qualifiedName}<${fieldType.markerName}>${
-                renderNullability(
-                    fieldType.nullable
-                )
-                }"
+                "$DataFrame<${fieldType.markerName}>${renderNullability(fieldType.nullable)}"
         }
 }
 
 internal object ShortNames : TypeRenderingStrategy {
-    override fun renderRowTypename(markerName: String): String {
-        return "${DataRow::class.simpleName}<${markerName.shorten()}>"
-    }
 
-    override fun renderDfTypename(markerName: String): String {
-        return "${ColumnsContainer::class.simpleName}<${markerName.shorten()}>"
-    }
+    private val DataRow = DataRow::class.simpleName!!
+    private val ColumnsContainer = ColumnsContainer::class.simpleName!!
+    private val DataFrame = DataFrame::class.simpleName!!
+    private val DataColumn = DataColumn::class.simpleName!!
+    private val ColumnGroup = ColumnGroup::class.simpleName!!
+
+    override fun renderRowTypename(markerName: String): String = "$DataRow<${markerName.shorten()}>"
+
+    override fun renderDfTypename(markerName: String): String = "$ColumnsContainer<${markerName.shorten()}>"
 
     override fun BaseField.renderColumnType(): Code =
         when (val fieldType = fieldType) {
-            is FieldType.ValueFieldType -> "${DataColumn::class.simpleName}<${fieldType.typeFqName.shorten()}>"
-            is FieldType.GroupFieldType -> "${ColumnGroup::class.simpleName}<${fieldType.markerName}>"
+            is FieldType.ValueFieldType ->
+                "$DataColumn<${fieldType.typeFqName.shorten()}>"
+
+            is FieldType.GroupFieldType ->
+                "$ColumnGroup<${fieldType.markerName}>"
+
             is FieldType.FrameFieldType ->
-                "${DataColumn::class.simpleName}<${DataFrame::class.simpleName}<${fieldType.markerName}>${
-                renderNullability(
-                    fieldType.nullable
-                )
-                }>"
+                "$DataColumn<$DataFrame<${fieldType.markerName}>${renderNullability(fieldType.nullable)}>"
         }
 
-    override fun BaseField.renderFieldType(): Code =
+    override fun BaseField.renderFieldType(asAccessor: Boolean): Code =
         when (val fieldType = fieldType) {
-            is FieldType.ValueFieldType -> fieldType.typeFqName.shorten()
-            is FieldType.GroupFieldType -> "${DataRow::class.simpleName}<${fieldType.markerName}>"
+            is FieldType.ValueFieldType ->
+                fieldType.typeFqName.shorten()
+
+            is FieldType.GroupFieldType ->
+                if (asAccessor) "$DataRow<${fieldType.markerName}>"
```

Comment:
Similar conversion happens for FrameFieldType and List. You can generate List<fieldType.markerName> for usages inside data schema declaration and DataFrame<>... for usages in accessors. What do you think about changing this argument to an enum UseSite { DataSchemaDeclaration, GeneratedAccessor }? Or even splitting in two functions. Took me quite a while to understand what's this about

Reply:
2 functions instead of 1 boolean argument is always better indeed :) 
Where is this conversion in lists you talk about?

Reply:
Okay, check out the latest commit, there I split the methods, renamed a bit and added explanation+examples :) should be a lot clearer what's happening now!

Reply:
I mean
@DataSchema
interface Name {
    val firstName: String
}   

@DataSchema
interface Departments {
    val people: List<Name> // Name is DataSchema, so accessor field type is DataFrame. 
    // same as
    val people: DataFrame<Name>
}

Reply:
Hmm, I think this is handled on the schema generation side already. `List<Name>` is a `ValueColumn` type. Can't really seem to find where the distinction happens with wrapping `DataRow` around it or not